context/context.go

```diff
@@ -24,12 +24,16 @@ import (
 	"sync"
 
 	"github.com/m3db/m3x/resource"
+
+	xnetcontext "golang.org/x/net/context"
```

Comment:
Why not use the [standard library package](https://golang.org/pkg/context/)?

Reply:
Wanted to be consistent with what we use in m3db

Reply:
They are type aliases for one another, you can use the std lib and be consistent.

Reply:
```go
package context

import "context" // standard library's context, as of Go 1.7

// A Context carries a deadline, a cancelation signal, and other values across
// API boundaries.
//
// Context's methods may be called by multiple goroutines simultaneously.
type Context = context.Context
```